sqlmesh/core/console.py

```diff
@@ -176,7 +184,45 @@ def stop_cleanup(self, success: bool = True) -> None:
         """
 
 
-class Console(LinterConsole, StateExporterConsole, StateImporterConsole, JanitorConsole, abc.ABC):
+class EnvironmentsConsole(abc.ABC):
+    """Console for displaying environments"""
+
+    @abc.abstractmethod
+    def print_environments(self, environments_summary: t.Dict[str, int]) -> None:
+        """Prints all environment names along with expiry datetime."""
+
+
+class DifferenceConsole(abc.ABC):
+    """Console for displaying environment differences"""
+
+    @abc.abstractmethod
+    def show_environment_difference_summary(
+        self,
+        context_diff: ContextDiff,
+        no_diff: bool = True,
+    ) -> None:
+        """Displays a summary of differences for the environment."""
+
+    @abc.abstractmethod
+    def show_model_difference_summary(
+        self,
+        context_diff: ContextDiff,
+        environment_naming_info: EnvironmentNamingInfo,
+        default_catalog: t.Optional[str],
+        no_diff: bool = True,
+    ) -> None:
+        """Displays a summary of differences for the given models."""
+
+
+class Console(
+    LinterConsole,
```

Comment:
in general multi inheritance is a bad pattern in python

Reply:
These are all interfaces. The methods don't have implementation so it doesn't really matter